app/models/goods_nomenclature.rb

```diff
@@ -131,6 +131,34 @@ def non_classifieds
     def non_grouping
       where(producline_suffix: '80')
     end
+
+    def with_footnotes
+      association_right_join(:footnotes)
+        .exclude(goods_nomenclatures__goods_nomenclature_item_id: nil)
+    end
+
+    def with_footnote_type_id(footnote_type_id)
+      return self if footnote_type_id.blank?
+
+      where(footnotes__footnote_type_id: footnote_type_id)
```

Comment:
do we need this kind of method?

is not ` where(footnotes__footnote_type_id: footnote_type_id)` self explanatory

Reply:
So the `return self` bit preserves the default dataset when the user supplied value is null. It means I can easily chain these methods together to filter in based on the inputs the user provides